source/extensions/filters/http/ext_proc/processor_state.h

```diff
@@ -205,7 +213,13 @@ class ProcessorState : public Logger::Loggable<Logger::Id::ext_proc> {
   // Flag to track whether Envoy already received the new timeout message.
   // Envoy should receive at most one such message in one particular state.
   bool new_timeout_received_{false};
+
+  // The buffer which stores all the received data that had not been shipped.
+  Buffer::OwnedImpl received_data_;
```

Comment:
Where is data that is delivered (shipped?) but not yet responded to kept?

Reply:
This received_data_ stores all the data received by the ext_proc filter. These data originally are stored in the different chunks of the queue: https://github.com/envoyproxy/envoy/blob/0fdaa31ec597a46ae7c26dc0ec736ba23a41d251/source/extensions/filters/http/ext_proc/processor_state.h#L32 

The data which had been delivered to the ext_proc server and pending for response is stored here.   After response is received, they will be shipped to either downstream client or upstream server.